src/login/logind-session-dbus.c

```diff
@@ -830,6 +839,22 @@ int session_send_create_reply(Session *s, sd_bus_error *error) {
         if (fifo_fd < 0)
                 return fifo_fd;
 
+        if (s->leader.fd >= 0) {
+                int r;
+
+                if (!s->leader_pidfd_event_source) {
+                        r = sd_event_add_io(s->manager->event, &s->leader_pidfd_event_source, s->leader.fd, EPOLLIN, session_dispatch_leader_pidfd, s);
+                        if (r < 0)
+                                return r;
+
+                        r = sd_event_source_set_priority(s->leader_pidfd_event_source, SD_EVENT_PRIORITY_IMPORTANT);
+                        if (r < 0)
+                                return r;
+
+                        (void) sd_event_source_set_description(s->leader_pidfd_event_source, "session-pidfd");
+                }
+        }
```

Comment:
please move this logic to a new helper session_watch_pidfd() that is part of logind-session.c